src/objective-c/GRPCClient/GRPCCall.m

```diff
@@ -139,8 +146,10 @@ - (void)finishWithError:(NSError *)errorOrNil {
   _self = nil;
 
   // If there were still request messages coming, stop them.
-  _requestWriter.state = GRXWriterStateFinished;
-  _requestWriter = nil;
+  @synchronized(_requestWriter) {
```

Comment:
`@synchronized(_requestWriter) {...}` with non-synchronized `_requestWriter = nil;` in another thread could have the first thread reading `self->_requestWriter` (to know what to lock on) concurrently with the second thread writing it, and those two things aren't atomic with ARC, are they?